src/client/components/filter-menu/time-filter-menu/fixed-time-tab.tsx

```diff
@@ -40,46 +50,83 @@ export interface FixedTimeTabProps {
 }
 
 export interface FixedTimeTabState {
-  start: Date;
-  end: Date;
+  start?: Date;
+  end?: Date;
   shift: string;
+  activeMonthStartDate?: Date;
+  hoverTimeRange?: TimeRange;
+  selectionSet?: boolean;
+  dateRangePickers: List<any>;
+  timeRanges: List<any>;
+  numTimeRanges: int;
 }
 
 export class FixedTimeTab extends React.Component<FixedTimeTabProps, FixedTimeTabState> {
-
   initialState = (): FixedTimeTabState => {
     const { essence, timekeeper, dimension: { name } } = this.props;
     const shift = essence.timeShift.toJS();
-
     const timeFilter = essence.getEffectiveFilter(timekeeper).clauseForReference(name);
     if (timeFilter && timeFilter instanceof FixedTimeFilterClause && !timeFilter.values.isEmpty()) {
-      const { start, end } = timeFilter.values.get(0);
-      return { start, end, shift };
+      if (timeFilter.values.size === 1) {
+        const { start, end } = timeFilter.values.get(0);
+        return { start, end, shift, activeMonthStartDate: null, hoverTimeRange: null, selectionSet: false, dateRangePickers: List(), timeRanges: List(), numTimeRanges: 1 };
+      } else {
+        const { start, end } = timeFilter.values.get(0);
+        let dateRanges = List();
+        for (let i = 0; i < timeFilter.values.size; i++) {
+          dateRanges = dateRanges.push([timeFilter.values.get(i).start, timeFilter.values.get(i).end]);
+        }
+        return { start, end, shift, activeMonthStartDate: null, hoverTimeRange: null, selectionSet: false,
+          dateRangePickers: List(), timeRanges: dateRanges, numTimeRanges: timeFilter.values.size };
+      }
     }
-    return { start: null, end: null, shift };
+    return { start: null, end: null, shift, activeMonthStartDate: null, hoverTimeRange: null, selectionSet: false, dateRangePickers: List(), timeRanges: List(), numTimeRanges: 1 };
   }
 
-  onStartChange = (start: Date) => this.setState({ start });
+  componentWillMount() {
+    const { start } = this.state;
+    const { essence: { timezone } } = this.props;
+    const flooredStart = month.floor(start || new Date(), timezone);
+    this.setState({
+      activeMonthStartDate: flooredStart,
+      selectionSet: true
+    });
+  }
 
-  onEndChange = (end: Date) => this.setState({ end });
+  onStartChange = (start: Date, index: int) => {
+    const { timeRanges, numTimeRanges } = this.state;
+    if (numTimeRanges === 1) {
```

Comment:
See, every time you have conditions on this value and different behaviour that should be unified